source/simulator/core.cc

```diff
@@ -657,7 +657,8 @@ namespace aspect
     // if the set of constraints has changed. If it did, we need to update the sparsity patterns.
     AffineConstraints<double> new_current_constraints;
     new_current_constraints.clear ();
-    new_current_constraints.reinit (introspection.index_sets.system_relevant_set);
+    new_current_constraints.reinit (dof_handler.locally_owned_dofs(),
+                                    introspection.index_sets.system_relevant_set);
```

Comment:
As @tjhei already noted, you'll need a `#if` around this. Or you go with the old function signature -- it may be marginally less efficient, but it's going to stick around for a couple more versions.